cloudwatch_monitoring_role/main.tf

```diff
@@ -0,0 +1,21 @@
+data "aws_iam_policy_document" "cloudwatch_monitoring_policy" {
+  statement {
+    sid     = "CloudWatchMonitoring"
+    actions = ["cloudwatch:PutMetricData", "cloudwatch:GetMetricStatistics", "cloudwatch:ListMetrics", "ec2:DescribeTags"]
+    effect  = "Allow"
+
+   resources = ["*"]
+  }
+}
+
+resource "aws_iam_policy" "cloudwatch_monitoring_policy" {
+  name   = "cloudwatch_${var.project}_${var.app}_${var.environment}"
```

Comment:
@minniux you are using `project`, `app` and `environment` here to build the name. I would enforce it to have these values passed instead of having empty string values as defaults.